webapp/controller/ListReport.controller.js

```diff
@@ -322,10 +348,68 @@ sap.ui.define([
 			const oSource = oEvent.getSource();
 			const oCtx = oSource.getBindingContext();
 			const oComponent = this.getOwnerComponent();
-
+			
 			oComponent.getRouter().navTo("ProductDetails", {
 				productId: oCtx.getObject("Id")
 			})
+		},
+
+		onCreateButtonPress: function() {
+			const sProductId = "newProduct";
+			const oComponent = this.getOwnerComponent();
+			oComponent.getRouter().navTo("ProductDetails", {
+				productId: sProductId
+			})	
+		},
+		
+		rewriteProductsIds: function() {
+			const aProducts = this.getView().getModel().getProperty("/Products");
+			aProducts.map((el, id) => {
```

Comment:
Don't use map because it creates a new array. 

If you need to do a side effect based on array and you don't want to create a new one, forEach is your friend